# Phylogenetic Tree Assembly and Updation Workflow

In this research, I propose two novel alignment-free, pairwise distance calculation methods based on k-mers and locality-sensitive hashing. Additionally, I have introduced a machine learning-based mechanism for phylogenetic tree construction. These innovative approaches aim to enhance the efficiency and accuracy of genetic distance calculations.

The proposed tree construction method utilizes a modified version of k-medoids, which demonstrates significantly improved performance compared to traditional phylogenetic tree construction techniques. As the final aspect of this research, I developed a numerical neural network to efficiently update the phylogenetic tree.

In summary, this work presents  methodologies for genetic distance calculation, phylogenetic tree construction, and dynamic tree updating, collectively advancing the field of computational phylogenetics.


## Introduction

The phylogenetic tree, also known as an evolutionary tree, is a branching diagram that illustrates the evolutionary relationships among various organisms. It organizes species based on genetic similarities and calculates their genetic distances to define these relationships. Phylogenetic trees serve as a foundational component in many areas of bioinformatics research, providing critical insights into evolutionary biology, comparative genomics, and molecular phylogenetics.

![Phylogenetic-Tree-Construction](https://raw.githubusercontent.com/ngimhana/Phylogenetic_tree_construction/master/Diagram/phylogenetic-tree.png)


## Methodology

![Phylogenetic-Tree-Construction-methodology](https://raw.githubusercontent.com/ngimhana/Phylogenetic_tree_construction/master/Diagram/mehodology.png)
